sstables/compress.cc

```diff
@@ -335,39 +335,69 @@ compression::locate(uint64_t position, const compression::segmented_offsets::acc
 
 }
 
-template <ChecksumUtils ChecksumType>
+// For SSTables 2.x (formats 'ka' and 'la'), the full checksum is a combination of checksums of compressed chunks.
+// For SSTables 3.x (format 'mc'), however, it is supposed to contain the full checksum of the file written so
+// the per-chunk checksums also count.
+enum class compressed_checksum_mode {
+    checksum_chunks_only,
+    checksum_all,
+};
+
+template<bool check_digest>
+struct digest_members {
+    uint32_t expected_digest;
+    uint32_t actual_digest;
+};
```

Comment:
I'm sure I saw this one patch ago. Is this in a different namespace? Why the duplication?

Reply:
>I'm sure I saw this one patch ago. Is this in a different namespace?

That's indeed in a different namespace. `compress.cc` leaves local symbols outside of the `sstables` namespace, whereas in `checksummed_data_source.cc` I've included everything under `sstables`.

>Why the duplication?

I thought it would be an overkill to add a new header just for this. None of the existing headers seem like a good fit.

Reply:
Ok. We aren't shy of adding tiny header files in general.